Add {pkgsMusl,pkgsStatic}.stdenv to build on hydra

domenkozar opened the pull request:
We've broken Musl stdenv quite a few times, so it makes sense to make build it and block the channel.

Musl has become quite mainstream and more and more things depend on it.

I've hit this since I'd like for patchelf to build against musl on CI for regression testing.

Fixes #81137

domenkozar committed:
Add pkgsMusl.stdenv to block nixpkgs channel

FRidh commented:
Then I think we should do the same for `pkgsStatic.stdenv`, since `pkgsStatic` has been getting quite the attention.

Ericson2314 commented:
This looks good to me. I had some branch somewhere tweaking `release-lib` so I think cross platforms and native platforms would looks more disimilar (I don't really like using `pkgCross`, `pkgStatic`, etc. in production), but we can worry about that later.

matthewbauer commented on pkgs/top-level/release.nix:
```diff
@@ -72,6 +72,9 @@ let
               jobs.transmission-gtk.x86_64-darwin
 
+              # Musl
+              jobs.pkgsMusl.stdenv.x86_64-linux
```
two things:

- this is in the Darwin-tested jobset, probably should be the nixos one.
- I don't think pkgsMusl is in jobs at all right now, you will need to add it like we do in haskell (https://github.com/NixOS/nixpkgs/pull/89425/files#diff-3f624960077266649716a0472a723f24L181)

matthewbauer replied:
but after doing (2), it will be in the jobset without being an actual blocker

FRidh commented:
Who will looking at fixing the builds in case a channel blocks? I am a bit hesitant making them blocking unless there multiple people invested in maintaining and fixing on a short notice.

domenkozar commented:
I've also added `pkgsStatic` and removed both as a blocker for the channel (I really just want Hydra to build these things).

About maintenance I think it's not clear for the most part of channel blockers, but I'd prefer not going into that right here :)

domenkozar commented:
@matthewbauer any ideas about https://gist.github.com/GrahamcOfBorg/9d6aad624d09b2798a5674b24c8a60db

domenkozar commented:
```$ nix-instantiate pkgs/top-level/release.nix
error: value is a set while a Boolean was expected, at undefined position
```

:exploding_head:

matthewbauer committed:
Don’t include i686-linux for pkgs{Musl,Static}

musl bootstrapping is unavailable for i686-linux right now. so we can
just exclude it from hydra.

matthewbauer commented:
> @matthewbauer any ideas about https://gist.github.com/GrahamcOfBorg/9d6aad624d09b2798a5674b24c8a60db

we don't have musl bootstrap-tools for i686-linux, so that's what this error is about.

domenkozar commented:
Thanks!

expipiplus1 commented:
It's really saddening when fixing some things in pkgsStatic to run across old commits fixing packages which have since become broken.

If blocking channels isn't an option, is there some middle ground to stop this package set bitrotting again and again? Perhaps when packages become broken at least some record could be made of the commit which changed things?

domenkozar commented:
I think the way forward is to have them block, but to come up with a team willing to maintain these.

expipiplus1 commented:
I'd be happy to help out and see how it goes

domenkozar commented:
@expipiplus1 could you open an issue and let's gather 3 people and start this effort.

expipiplus1 commented:
Done: https://github.com/NixOS/nixpkgs/issues/103290
